### ISSUE TITLE
Convert Georges Editor to be an OVQt Plugin

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#11]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/11) by Matt Raykowski (Bitbucket: [Matt Raykowski](https://bitbucket.org/Matt Raykowski), ).**

----------------------------------------

Convert the Georges Editor MFC tool to be a plugin to OVQt.

**Comment by ryzom-pipeline** [comment]
**Original comment by aquiles (Bitbucket: [aquiles](https://bitbucket.org/aquiles), GitHub: [aquiles](https://github.com/aquiles)).**

----------------------------------------

MFC? Not Qt? :(

**Comment by ryzom-pipeline** [comment]
**Original comment by Matt Raykowski (Bitbucket: [Matt Raykowski](https://bitbucket.org/Matt Raykowski), ).**

----------------------------------------

Maybe that's poorly worded. We're converting the old MFC tool to OVQt which, implicitly, is Qt.

I have a substantial portion of the rework done in *branch-features-geqt-plugin*. I had to deviate quite a bit from the way you were doing things because of limitations in looking up types. You saw some of the symptoms when you loaded sheets and it spewed a ton of warnings.

You took the sheet and look at the type of the variable in there and polled it from parents/defaults if available. The problem really appeared when I implemented the ability to create new sheets, it had no data for the fields I wanted to add and wouldc rash. The MFC-based tool does the complete inverse of your way - instead of looking at the sheet it determines the DFNs involve and loads them. This is part of the reason why when you load a new sheet up in the old editor it shows you every single available element. We can be smart and still have the features you had but we need to initialize the form the way the old tool does.
